agent/agentcontainers/api.go

```diff
@@ -315,12 +317,26 @@ func NewAPI(logger slog.Logger, options ...Option) *API {
 		api.subAgentClient.Store(&c)
 	}
 
-	go api.watcherLoop()
-	go api.updaterLoop()
+	go func() {
+		select {
+		case <-api.ctx.Done():
+			break
+		case <-api.initialized:
+			go api.watcherLoop()
+			go api.updaterLoop()
```

Comment:
An important consideration here is that `Close` is dependent on these functions closing their channels.